### PR TITLE
[dagster-airbyte] Enable setting a destination table prefix using  managed ingestion

### PR DESCRIPTION
**Pull request by benpankow**
## Summary

Enables setting the [destination `prefix` key of an Airbyte connection](https://docs.airbyte.com/understanding-airbyte/connections/#prefix-stream-name), which prefixes all tables created in the destination.

Ex:

```python
my_prefixed_connection = AirbyteConnection(
    name="local-json-conn",
    source=...,
    destination=...,
    stream_config=...,
    prefix="my_prefix",
)
```

## Test Plan

New integration test using prefixed connections.

### CODE REVIEW

**Comment by benpankow** [comment]
Current dependencies on/for this PR:
* master
  * **PR #11057** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11057" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/11057?utm_source=stack-comment).

File: python_modules/libraries/dagster-airbyte/dagster_airbyte/managed/types.py [hunk 1]
```diff
@@ -353,6 +356,7 @@ def from_api_json(
                 destination_namespace=api_dict["namespaceFormat"]
                 if api_dict["namespaceDefinition"] == "customformat"
                 else AirbyteDestinationNamespace(api_dict["namespaceDefinition"]),
+                prefix=api_dict["prefix"] if api_dict["prefix"] else None,
```
Comment (rexledesma) [inline python_modules/libraries/dagster-airbyte/dagster_airbyte/managed/types.py:359]: terseness
```suggestion
                prefix=api_dict.get("prefix"),
```

File: python_modules/libraries/dagster-airbyte/dagster_airbyte/managed/types.py [hunk 2]
```diff
@@ -287,6 +289,7 @@ def __init__(
         self.destination_namespace = check.opt_inst_param(
             destination_namespace, "destination_namespace", (str, AirbyteDestinationNamespace)
         )
+        self.prefix = check.opt_str_param(prefix, "prefix")
```
Comment (rexledesma) [inline python_modules/libraries/dagster-airbyte/dagster_airbyte/managed/types.py:292]: Is there any validation that we need to apply to the prefix?

**Review by rexledesma — APPROVED** [review]

**Comment by blarghmatey** [comment]
This is great! Any thoughts on when this might land? I'm currently managing my Airbyte connections manually and can't wait to get them under code with Dagster :)

**Comment by benpankow** [comment]
> This is great! Any thoughts on when this might land? I'm currently managing my Airbyte connections manually and can't wait to get them under code with Dagster :)

Hi Tobias, we will try to get this in with this week's release - sorry for the delay!

**Pull request merged**